src/pages/workspace/reimburse/WorkspaceReimburseView.js

```diff
@@ -64,140 +65,142 @@ const defaultProps = {
     reimbursementAccount: ReimbursementAccountProps.reimbursementAccountDefaultProps,
 };
 
-class WorkspaceReimburseView extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            currentRatePerUnit: this.getCurrentRatePerUnitLabel(),
-        };
-    }
-
-    componentDidMount() {
-        this.fetchData();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.policy.customUnits !== this.props.policy.customUnits || prevProps.preferredLocale !== this.props.preferredLocale) {
-            this.setState({currentRatePerUnit: this.getCurrentRatePerUnitLabel()});
-        }
-
-        const reconnecting = prevProps.network.isOffline && !this.props.network.isOffline;
-        if (!reconnecting) {
-            return;
-        }
+function WorkspaceReimburseView(props) {
+    const [currentRatePerUnit, setCurrentRatePerUnit] = useState('');
+    const prevIsOffline = usePrevious(props.network.isOffline);
+    const viewAllReceiptsUrl = `expenses?policyIDList=${props.policy.id}&billableReimbursable=reimbursable&submitterEmail=%2B%2B`;
+    const distanceCustomUnit = _.find(lodashGet(props.policy, 'customUnits', {}), (unit) => unit.name === 'Distance');
+    const distanceCustomRate = _.find(lodashGet(props.distanceCustomUnit, 'rates', {}), (rate) => rate.name === 'Default Rate');
+
+    const getNumericValue = useCallback(
+        (value) => {
+            const propsToLocaleDigit = props.toLocaleDigit;
```

Comment:
Do you know if we need to do this? If not, I'll suggest using directly `props.toLocaleDigit`.

Reply:
Yes we need to destructure the props else it will give eslint warning for exhaustive deps
You can see discussions [over here](https://github.com/facebook/react/issues/16265)

Reply:
@KrAbhas I think the proper destructing of the props is outside the `useCallback` or `useEffect`. For example here

https://github.com/Expensify/App/blob/60f48175c9e3342ef192d878218820e5ed5a76cd/src/components/MoneyRequestConfirmationList.js#L101-L104

Reply:
Okay, we can do it like this, but I wonder what may be the issue with destructing inside a these hooks. Do you see or anticipate any logical issue?

Reply:
There's no problem at all; it's just a tiny readability issue and we don't have a proper strategy for this kind of issue so we just need to follow what we did.

Reply:
Yeah that's understandable. I have made the changes.